### PR TITLE
Fix contentType set properly with putObject, fPutObject.

### LINKED ISSUE #530: putObject ignore contentType
putObject always upload as 'application/octet-stream'

for example
```JS
var _100kb = new Buffer(100*1024)
_100kb.fill('a')
minioClient.putObject('data', 'test.png', _100kb, 'image/png', function(err, etag) {})
```
test.png in server is 'application/octet-stream' not 'image/png'

### PR DESCRIPTION
**Pull request by harshavardhana**
Fixes #530

### CODE REVIEW

File: src/main/minio.js [hunk 1]
```diff
@@ -819,15 +820,23 @@ export class Client {
...
-      contentType = 'application/octet-stream'
+    if (isFunction(contentType)) {
+      callback = contentType
+    }
+
+    if (!mime.lookup(filePath)) {
+        contentType = 'application/octet-stream'
+    } else {
+        contentType = mime.lookup(filePath)
+    }
```
Comment (krishnasrinivas) [inline src/main/minio.js]: the contentType argument value that is passed to the function is lost.

Comment (harshavardhana) [inline src/main/minio.js]: Oops yes

**Comment by harshavardhana** [comment]
done @krishnasrinivas

File: src/main/minio.js [hunk 2]
```diff
@@ -819,15 +820,25 @@ export class Client {
...
+
     if (!isString(filePath)) {
       throw new TypeError('filePath should be of type "string"')
     }
 
+    if (isFunction(contentType)) {
+      callback = contentType
+    }
+
     if (contentType.trim() === '') {
```
Comment (krishnasrinivas) [inline src/main/minio.js]: if contentType is a function then this statement will error

**Pull request merged**